lib/command/terraform/generate.rb

```diff
@@ -5,26 +5,82 @@ module Terraform
     class Generate < Base
       SUBCOMMAND_NAME = "terraform"
       NAME = "generate"
+      OPTIONS = [
+        app_option(required: false),
+        dir_option(required: false)
+      ].freeze
       DESCRIPTION = "Generates terraform configuration files"
       LONG_DESCRIPTION = <<~DESC
         - Generates terraform configuration files based on `controlplane.yml` and `templates/` config
       DESC
       WITH_INFO_HEADER = false
-      VALIDATIONS = [].freeze
 
       def call
-        File.write(terraform_dir.join("providers.tf"), cpln_provider.to_tf)
+        generate_common_configs
+        generate_app_configs
       end
 
       private
 
-      def cpln_provider
-        TerraformConfig::RequiredProvider.new("cpln", source: "controlplane-com/cpln", version: "~> 1.0")
+      def generate_common_configs
+        cpln_provider = TerraformConfig::RequiredProvider.new(
+          "cpln",
+          source: "controlplane-com/cpln",
+          version: "~> 1.0"
+        )
+
+        File.write(terraform_dir.join("providers.tf"), cpln_provider.to_tf)
+      end
+
+      def generate_app_configs
+        Array(config.app || config.apps.keys).each do |app|
+          generate_app_config(app.to_s)
+        end
+      end
+
+      def generate_app_config(app)
+        config.class.define_method(:app) { app }
+
+        terraform_app_dir = recreate_terraform_app_dir(app)
+
+        templates.each do |template|
+          generator = TerraformConfig::Generator.new(config: config, template: template)
+
+          # TODO: Delete line below after all template kinds are supported
+          next unless %w[gvc identity].include?(template["kind"])
+
+          File.write(terraform_app_dir.join(generator.filename), generator.tf_config.to_tf, mode: "a+")
+        end
+      end
+
+      def recreate_terraform_app_dir(app_path)
+        full_path = terraform_dir.join(app_path)
+
+        FileUtils.rm_rf(full_path)
```

Comment:
We need a check to make sure this rm -rf never goes beyond the project dir.

Reply:
Done